packages/cli-platform-ios/src/commands/runIOS/index.ts

```diff
@@ -32,11 +41,43 @@ export interface FlagsT extends BuildFlags {
   udid?: string;
   binaryPath?: string;
   listDevices?: boolean;
+  packager?: boolean;
+  port: number;
+  terminal?: string;
 }
 
 async function runIOS(_: Array<string>, ctx: Config, args: FlagsT) {
   link.setPlatform('ios');
 
+  let {packager, port} = args;
+
+  const packagerStatus = await isPackagerRunning(port);
+
+  if (
+    typeof packagerStatus === 'object' &&
+    packagerStatus.status === 'running'
+  ) {
+    if (packagerStatus.root === ctx.root) {
+      packager = false;
+      logAlreadyRunningBundler(port);
+    } else {
+      const result = await handlePortUnavailable(port, ctx.root, packager);
+      [port, packager] = [result.port, result.packager];
+    }
+  } else if (packagerStatus === 'unrecognized') {
+    const result = await handlePortUnavailable(port, ctx.root, packager);
+    [port, packager] = [result.port, result.packager];
+  }
+
+  if (packager) {
+    await startServerInNewWindow(
```

Comment:
await causes the CLI to hang indefinitely. Must be called without await. 
Check existing comment at line no 105